update readme (#303)
ingonyama-zk/icicle: wrappers/rust/README.md

### Rust Bindings

For convenience, we also provide rust bindings to the ICICLE library for the following primitives:

- MSM
- NTT
    - Forward NTT
    - Inverse NTT

> - ECNTT

Reviewer: lines 9-13 aren't currently supported from Rust, maybe best to remove them for now

    - Forward ECNTT
    - Inverse NTT
- Scalar Vector Multiplication
- Point Vector Multiplication

> A custom [build script][B_SCRIPT] is used to compile and link the ICICLE library. The environment variable `ARCH_TYPE` is used to determine which GPU type the library should be compiled for and it defaults to `native` when it is not set allowing the compiler to detect the installed GPU type.

Author: oh i just didnt update it yet. please look now.

> NOTE: A GPU must be detectable and therefore installed if the `ARCH_TYPE` is not set.

Once you have your parameters set, run:

```sh
cargo build --release
```

> You'll find a release ready library at `target/release/libicicle_utils.rlib`.

Reviewer 2: I think the lib name has changed

Author: i just didnt update it yet. please look now.

To benchmark and test the functionality available in RUST, run:

```

> cargo bench

Reviewer: benches also not supported rn :)

cargo test -- --test-threads=1
```

> The flag `--test-threads=1` is needed because currently some tests might interfere with one another inside the GPU.

Reviewer: shouldn't be needed anymore

### Rust Bindings

`icicle-core` defines all interfaces, macros and common methods.

`icicle-cuda-runtime` defines `DeviceContext` which can be used to manage a specific GPU as well as wrapping common CUDA methods.

> `icicle-curves` implement all interfaces an macros from `icicle-core` for each curve. For example `icicle-bn254` implements curve `bn254`. Each curve has its own [build script](./icicle-curves/icicle-bn254/build.rs) which will build the CUDA libraries for that curve as part of the rust-toolchain build.

Reviewer 2: ```suggestion
`icicle-curves` implements all interfaces and macros from `icicle-core` for each curve. For example `icicle-bn254` implements curve `bn254`. Each curve has its own [build script](./icicle-curves/icicle-bn254/build.rs) which will build the CUDA libraries for that curve as part of the rust-toolchain build.
```

## Building a curve and running tests

Enter a curve implementation.

```
cd icicle-curves/icicle-bn254
```

To build 

```sh
cargo build --release
```

The build may take a while because we are also building the CUDA libraries for the selected curve.

To run benchmarks

```
cargo bench
```

To run test

```sh
cargo test -- --test-threads=1
```

The flag `--test-threads=1` is needed because currently some tests might interfere with one another inside the GPU.

### Example Usage

An example of using the Rust bindings library can be found in our [fast-danksharding implementation][FDI]

<!-- Begin Links -->
[FDI]: https://github.com/ingonyama-zk/fast-danksharding

> <!-- End Links -->

Reviewer 2: Maybe remove this? It does not use the latest API
